src/form/password-input/PasswordInput.tsx

```diff
@@ -0,0 +1,83 @@
+import "./_password-input.scss";
+
+import React, {useState} from "react";
+import classNames from "classnames";
+
+import Input from "../input/Input";
+import Button from "../../button/Button";
+
+export interface PasswordInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  onChange: React.ReactEventHandler<HTMLInputElement>;
+  onFocus?: React.ReactEventHandler<HTMLInputElement>;
+  onBlur?: React.ReactEventHandler<HTMLInputElement>;
+  onKeyDown?: React.KeyboardEventHandler<HTMLInputElement>;
+  onKeyUp?: React.KeyboardEventHandler<HTMLInputElement>;
+  onInput?: React.KeyboardEventHandler<HTMLInputElement>;
+  passwordVisibilityButtonTestId?: string;
+  hidePasswordButtonIcon?: Node;
+  showPasswordButtonIcon?: Node;
+  testid: string;
+  name: string;
+  value?: string;
+  hasError?: boolean;
+  placeholder?: string;
+  customClassName?: string;
+}
+
+function PasswordInput({
+  testid,
+  name,
+  value,
+  onChange,
+  hasError,
+  placeholder,
+  customClassName,
+  hidePasswordButtonIcon,
+  showPasswordButtonIcon,
+  passwordVisibilityButtonTestId
+}: PasswordInputProps) {
+  const [isPasswordShown, setPasswordVisibility] = useState(false);
+  const passwrodInputIconClassName = classNames("password-input__icon", {
```

Comment:
```suggestion
  const passwordInputIconClassName = classNames("password-input__icon", {
```
There is a small typo here